### DISCUSSION TITLE
Capture ALPV license plate read data using python?

### DISCUSSION THREAD

**Discussion by orionsparks** [AxisCommunications/discussions#210] [ACAP] [answered]
Does anyone have a code snippet that properly captures an ALPR license plate read POST request from a camera?

**Comment by shadysp** [comment]
Check the[ ALPV API](https://www.axis.com/vapix-library/subjects/t10102231/section/t10165701/display).
There are available examples for receiving data:
-[Using TCP.](https://www.axis.com/vapix-library/subjects/t10102231/section/t10165701/display?section=t10165701-t10165795)
-[Using HTTP.](https://www.axis.com/vapix-library/subjects/t10102231/section/t10165701/display?section=t10165701-t10165794)
-[PHP example of a receiver. ](https://www.axis.com/vapix-library/subjects/t10102231/section/t10165701/display?section=t10165701-t10165793)

**Comment by vivekatoffice** [comment, marked as answer]
Hi @orionsparks ,
Please go through the links provided by @shadysp in previous comment.
I am adding a very simple sample code in Python for receiving ALPV metadata over TCP:

# TCP

## Settings on ALPV page
![TCP settings](https://github.com/AxisCommunications/discussions/assets/52032125/3e5ee280-c8e6-426c-8523-8438b27b0239)

## Python code

```python
import socket
import json

HOST = "192.168.50.30"  # The server's hostname or IP address
PORT = 50000  # The port used by the server

def recvall(sock):
    BUFF_SIZE = 4096 # 4 KiB
    data = b''
    while True:
        part = sock.recv(BUFF_SIZE)
        data += part
        if len(part) < BUFF_SIZE:
            # either 0 or end of data
            break
    return data

with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
    s.bind((HOST, PORT))
    s.listen()
    while True:
        conn, addr = s.accept()
        print(f"Connected by {addr}")
        data = recvall(conn)
        parsed_data = json.loads(data)
        print(parsed_data)
        conn.close()            

```

## Output on console

![Output](https://github.com/AxisCommunications/discussions/assets/52032125/cd55dde5-0468-47f5-9667-df3908e74163)
